codec-http/src/main/java/io/netty/handler/codec/http/multipart/DeleteFileOnExitHook.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.multipart;
+
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Extend {@link java.io.DeleteOnExitHook} to dynamically manipulate hook records.
+ */
+public final class DeleteFileOnExitHook {
```

Comment:
Does this class really need to be public?

Reply:
At first I thought maybe it would be used elsewhere.

Reply:
please declare it package-private

Reply:
OK.